shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/impl/ShardingSpherePipelineDataSourceConfigurationTest.java

```diff
@@ -58,4 +60,28 @@ private String getDataSourceYaml() {
         yamlDataSourceConfigs.forEach((key, value) -> result.put(key, new YamlDataSourceConfigurationSwapper().swapToDataSourceProperties(value)));
         return result;
     }
+
+    @Test
+    public void assertDataSourceCanBeAggregation() throws InvalidDataSourcePropertiesException {
+        ShardingSpherePipelineDataSourceConfiguration dataSourceConfig = new ShardingSpherePipelineDataSourceConfiguration(getDataSourceAggregationYaml());
+        List<DataSourceProperties> actual = new ArrayList<>(getDataSourcePropertiesMap(dataSourceConfig.getRootConfig().getDataSources()).values());
```

Comment:
How about change ArrayList to LinkedList?

Reply:
Fix in https://github.com/apache/shardingsphere/pull/16710